### ISSUE TITLE
React: Accessibility improvements for Popup

### ISSUE THREAD

**Issue by anicholls** [Workday/canvas-kit#59]
## 🐛 Bug Report

`Popup` does not currently pass accessibility requirements. We cannot use it for many cases unless we add the ability to lock/cycle focus within them, and focus on the first focusable item when opened, etc.

We need to support everything in https://bitsofco.de/accessible-modal-dialog/ except for # 4:

> 1. Markup the Dialog and Dialog Overlay Appropriately
> 2. On Dialog Open, Set Focus
> 3. On Dialog Close, Return Focus to the Last Focused Element
> 5. While Open, Prevent Tabbing to Outside the Dialog
> 6. While Open?
> 7. Allow the ESC Key to Close the Dialog
